src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/TempMultipleAlignmentsReclassifier.java

```diff
@@ -0,0 +1,211 @@
+        //      If this unique read region is "short": shorter than {@code uniqReadLenInclusive}), drop it.
+
+        // each alignment has an entry of a tuple2, one for max overlap front, one for max overlap rear,
+        // max overlap front is a tuple2 registering the index and overlap bases count
+        final List<Tuple2<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>>> maxOverlapMap =
+                getMaxOverlapPairs(selectedAlignments);
+
+        final List<Integer> idxToRemove = new ArrayList<>(selectedAlignments.size());
+        for (int i = 0; i < selectedAlignments.size(); ++i) {
+            final AlignmentInterval cur = selectedAlignments.get(i);
+            final Tuple2<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>> maxOverlapFrontAndRear = maxOverlapMap.get(i);
+            final int maxOverlapFront = Math.max(0, maxOverlapFrontAndRear._1._2);
+            final int maxOverlapRead = Math.max(0, maxOverlapFrontAndRear._2._2);
+
+            final int uniqReadSpan = cur.endInAssembledContig - cur.startInAssembledContig + 1 - maxOverlapFront - maxOverlapRead;
+            if (uniqReadSpan < uniqReadLenInclusive)
+                idxToRemove.add(i);
+        }
+
+        if ( idxToRemove.isEmpty() )
+            return new Tuple2<>(selectedAlignments, lowUniquenessMappings);
+
+        // removing in reverse order so that iterators are not invalidated if we were to remove from start
+        final ListIterator<Integer> rit = idxToRemove.listIterator(idxToRemove.size());
+        while (rit.hasPrevious()) {
+            selectedAlignments.remove( rit.previous().intValue() );
+        }
+
+        return new Tuple2<>(selectedAlignments, lowUniquenessMappings);
+    }
+
+    /**
+     * Each alignment has an entry of a tuple2, one for max overlap front, one for max overlap rear,
+     * max overlap front is a tuple2 registering the index and overlap bases count
+     */
+    private static List<Tuple2<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>>> getMaxOverlapPairs(final List<AlignmentInterval> selectedAlignments) {
+
+        final List<Tuple2<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>>> maxOverlapMap =
+                new ArrayList<>(Collections.nCopies(selectedAlignments.size(), new Tuple2<>(new Tuple2<>(-1, -1), new Tuple2<>(-1, -1))));
+
+        for(int i = 0; i < selectedAlignments.size() - 1; ++i) {
+            final AlignmentInterval cur = selectedAlignments.get(i);
+            int maxOverlap = -1;
+            int maxOverlapIdx = -1;
+            for (int j = i + 1; j < selectedAlignments.size(); ++j) {
+                final int overlap = AlignmentInterval.overlapOnContig(cur, selectedAlignments.get(j));
+                if (overlap > 0) {
+                    maxOverlap = Math.max(maxOverlap, overlap);
+                    maxOverlapIdx = j;
+                }
+                else // following ones, as guaranteed by the ordering of alignments in the contig, cannot overlap
```

Comment:
I'm also a little confused about this logic. What if the first alignment doesn't overlap this one (say this one is no. 4) but alignments no. 2 and 3 do?